test/test-delegate.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const Test = require('tape');
+const GraphQLComponent = require('../lib');
+
+Test('component resolver delegate', async (t) => {
```

Comment:
these tests effectively test that the resolver delegate implementation allows `execute()` to work as expected/as it did were the resolver delegate change not added, but is there any way to test the negative case that this new feature prevents: type resolver execution duplication?

^ that's assuming I'm understanding all of this correctly, if I'm way off, please disregard.

Reply:
It isn't testing execute, it is just using execute as an entry point (instead of having to start a graphql server and send graphql requests).

Reply:
So what exactly does that test then?

Reply:
yeah that makes sense. It's just hard to tell what this is testing outside of what `test-execute.js` tests. I know these tests have a component that imports another, but the assertions look very much like the ones found in `test-execute.js`. Anyway we could dive into the `resolver.js` functions and get a little more in depth on testing the proxy functionality or is that overkill?

Reply:
Added more tests.